docs/activityadvanced.md

```diff
@@ -0,0 +1,137 @@
+so let's create a new state: "startled".
+<br>
+Let's add another if statement inside the
+<code class="language-plaintext highlighter-rouge">current_state()</code>
+function inside the
+<code class="language-plaintext highlighter-rouge">bird.py</code>  file.
+So, if there was a "loud" event, the function can return the `startled` state.
+<br>
+You can find more info about the microphone in the <a href="https://microbit-micropython.readthedocs.io/en/v2-docs/microphone.html" target="_blank">Microphone documentation</a>.
+</p>
+
+[Solution](https://github.com/microbit-carlos/microbit-bird-activity/tree/main/solutions/4-bird-startled.md)
+
+Now make a noise, like a clap, near the microphone and watch your bird get
+startled!
+
+Hint: If the conference area is too loud, you can blow on the microphone.
+
+
+## Sending a Hello radio message
+
+Our birds are reacting to events in their environment like another bird 
+saying `hello`, or seeing a `cat`.
+
+How does this work?
+
+The micro:bit ration is uses to broadcast and receive these messages.
+
+Let's look at the `warn_about_cat()` function towards the bottom of the
+`bird.py` file:
+
+```python
+def warn_about_cat():
+    # Impose a delay before warning others
+    __mb.sleep(__rand.randint(500,1500))
+    __radio.send("cat->all")
+```
+
+Here we can see two thing:
+
+1. A delay for a random amount of time between 500 milliseconds to 1.5 seconds.
+    - This is done so that there is a bit of delay between each bird warning
+      each other
+2. Sending the `cat->all` radio message
+
+<p class="exercise">
+Exercise: <br>
+Let's create a new function in
+<code class="language-plaintext highlighter-rouge">bird.py</code>
+that says "hello" to all the birds nearby. This will be very similar to the
+<code class="language-plaintext highlighter-rouge">warn_about_cat()</code>
+function.
+<br><br>
+Now, in the 
+<code class="language-plaintext highlighter-rouge">while True:</code>
+loop, let's call the new bird function when
+<code class="language-plaintext highlighter-rouge">button_a</code>
+is pressed, and you will be able to see the other birds react to your
+"Hello" message each time the button A is pressed.
+</p>
+
+Hint: You might need to add a small delay after saying hello, or you can end
+up sending a lot of hello messages around on a single button press.
+
+[Solution](https://github.com/microbit-carlos/microbit-bird-activity/tree/main/solutions/5-bird-hello-everyone.md)
+
+## Say hello to a specific neighbour
+
+The format of these radio messages is very simple: `event->target`
+- `event` is the name of the event
+- `target` is the "friendly name" of the micro:bit (unique to each board),
+  or "all" to send to everyone
+
+So you can send a "hello" message to a specific bird, but how do you find out
+the "friendly name" of a micro:bit? 
+
+<p class="exercise">
+Exercise: <br>
+The friendly name can be obtained using the `bird.friendly_name()` function.
```

Comment:
This is a lovely touch. Nice!

Reply:
@microbit-matt-hillsdon converted the C CODAL code into MicroPython for a support request, I just copied the code 😁